base/reducedim.jl

```diff
@@ -323,6 +323,7 @@ mapreduce(f, op, A::AbstractArrayOrBroadcasted; dims=:, init=_InitialValue()) =
     _mapreduce_dim(f, op, init, A, dims)
 mapreduce(f, op, A::AbstractArrayOrBroadcasted...; kw...) =
     reduce(op, map(f, A...); kw...)
+mapreduce(f, op, A::AbstractArrayOrBroadcasted...) = mapreduce(A->f(A...),op,zip(A...))
```

Comment:
This happens to work because of https://github.com/JuliaLang/julia/issues/9498, but we should not rely on that. We should implement a similar approach as the one-arg case above, where we call an internal function with positional arguments to be able to dispatch on the type of `dims`.